app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    can %i[index show], User
+    can %i[index new create show], Post
+    can %i[new create], Comment
+    can %i[update destroy], Post, { author_id: user.id }
+    can %i[update destroy], Comment, { author_id: user.id }
+    can :manage, Post if user.is? :admin
+    can :manage, Comment if user.is? :admin
```

Comment:
- [ ] you can actually refactor this to `   can :manage, :all if user.is? :admin` :+1: